tests/trainer/connectors/test_callback_connector.py

```diff
@@ -53,3 +73,69 @@ def test_all_callback_states_saved_before_checkpoint_callback(tmpdir):
     assert "content0" in state0 and state0["content0"] == 0
     assert "content1" in state1 and state1["content1"] == 1
     assert type(checkpoint_callback) in ckpt["callbacks"]
+
+
+def test_attach_model_callbacks():
+    """ Test that the callbacks defined in the model and through Trainer get merged correctly. """
+
+    def assert_composition(trainer_callbacks, model_callbacks, expected):
+        model = Mock()
+        model.configure_callbacks.return_value = model_callbacks
+        trainer = Trainer(checkpoint_callback=False, progress_bar_refresh_rate=0, callbacks=trainer_callbacks)
+        cb_connector = CallbackConnector(trainer)
+        cb_connector._attach_model_callbacks(model)
+        assert trainer.callbacks == expected
+
+    early_stopping = EarlyStopping()
+    progress_bar = ProgressBar()
+    lr_monitor = LearningRateMonitor()
+    grad_accumulation = GradientAccumulationScheduler({1: 1})
+
+    # no callbacks
+    assert_composition(trainer_callbacks=[], model_callbacks=[], expected=[])
+
+    # callbacks of different types
+    assert_composition(
+        trainer_callbacks=[early_stopping], model_callbacks=[progress_bar], expected=[early_stopping, progress_bar]
+    )
+
+    # same callback type twice, different instance
+    assert_composition(
+        trainer_callbacks=[progress_bar, EarlyStopping()],
+        model_callbacks=[early_stopping],
+        expected=[progress_bar, early_stopping]
+    )
+
+    # multiple callbacks of the same type in trainer
+    assert_composition(
+        trainer_callbacks=[LearningRateMonitor(),
+                           EarlyStopping(),
+                           LearningRateMonitor(),
+                           EarlyStopping()],
+        model_callbacks=[early_stopping, lr_monitor],
+        expected=[early_stopping, lr_monitor]
+    )
+
+    # multiple callbacks of the same type, in both trainer and model
+    assert_composition(
+        trainer_callbacks=[
+            LearningRateMonitor(), progress_bar,
+            EarlyStopping(),
+            LearningRateMonitor(),
+            EarlyStopping()
+        ],
```

Comment:
same here, for some reason there are two elements on one line and the rest on the others. I thought it is supposed to give consistent formatting

Reply:
yes, this looks strange...